handler.go

```diff
@@ -482,29 +521,76 @@ func (r *cacheRequest) isCacheable() bool {
 }
 
 func newResponseBuffer(w http.ResponseWriter) *responseBuffer {
+	strm, err := stream.NewStream("responseBuffer", stream.NewMemFS())
+	if err != nil {
+		panic(err)
+	}
 	return &responseBuffer{
 		ResponseWriter: w,
-		Buffer:         &bytes.Buffer{},
+		Stream:         strm,
+		//Buffer:         &bytes.Buffer{},
+		C: make(chan struct{}),
 	}
 }
 
 type responseBuffer struct {
```

Comment:
Perhaps worth removing Buffer entirely and calling it something different than `responseBuffer`?